pkg/morph/event/notary_preparator.go

```diff
@@ -141,15 +143,10 @@ func (p preparator) Prepare(nr *payload.P2PNotaryRequest) (NotaryEvent, error) {
 	}
 	invokerWitness := ln == 4
 
-	// alphabet node should handle only notary requests that do not yet have inner
-	// ring multisignature filled => such main TXs either have empty invocation script
-	// of the inner ring witness (in case if Notary Actor is used to create request)
-	// or have it filled with dummy bytes (if request was created manually with the old
-	// neo-go API)
-	//
-	// this check prevents notary flow recursion
-	if !(len(nr.MainTransaction.Scripts[1].InvocationScript) == 0 ||
-		bytes.Equal(nr.MainTransaction.Scripts[1].InvocationScript, p.dummyInvocationScript)) { // compatibility with old version
+	// We do receive requests made by the node itself, but these must never
+	// be reacted upon, otherwise we'll never end sending these requests
+	// (at least not until the main tx is ready).
+	if nr.FallbackTransaction.Signers[1].Account == p.localAcc {
```

Comment:
https://pkg.go.dev/github.com/nspcc-dev/neo-go@v0.105.0/pkg/util#Uint160.Equals is more preferable way when provided